EventVisualisation/Detectors/src/DataInterpreterITS.cxx

```diff
@@ -0,0 +1,174 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file DataInterpreterITS.cxx
+/// \brief converting ITS data to Event Visualisation primitives
+/// \author julian.myrcha@cern.ch
+/// \author p.nowakowski@cern.ch
+
+#include "EventVisualisationBase/ConfigurationManager.h"
+#include "EventVisualisationDetectors/DataInterpreterITS.h"
+#include "EventVisualisationDataConverter/VisualisationEvent.h"
+
+#include "DataFormatsITS/TrackITS.h"
+#include "DataFormatsITSMFT/Cluster.h"
+#include "DataFormatsITSMFT/ROFRecord.h"
+#include "ITSBase/GeometryTGeo.h"
+
+#include <TEveManager.h>
+#include <TEveTrackPropagator.h>
+#include <TEveTrack.h>
+#include <TGListTree.h>
+#include <TFile.h>
+#include <TTree.h>
+#include <TVector2.h>
+
+#include <gsl/span>
+
+using namespace std;
+
+namespace o2
+{
+namespace event_visualisation
+{
+
+DataInterpreterITS::DataInterpreterITS()
+{
+  //Prepare coordinate translator
+  base::GeometryManager::loadGeometry("O2geometry.root", "FAIRGeom");
+  its::GeometryTGeo* gman = its::GeometryTGeo::Instance();
+  gman->fillMatrixCache(o2::utils::bit2Mask(o2::TransformType::T2GRot));
+}
+
+std::unique_ptr<VisualisationEvent> DataInterpreterITS::interpretDataForType(TObject* data, EVisualisationDataType type)
+{
+  TList* list = (TList*)data;
+  Int_t event = ((TVector2*)list->At(2))->X();
+
+  auto ret_event = std::make_unique<VisualisationEvent>(0, 0, 0, 0, "", 0);
+
+  if (type == Clusters) {
+    its::GeometryTGeo* gman = its::GeometryTGeo::Instance();
+
+    TFile* clustFile = (TFile*)list->At(1);
+    TTree* clusters = (TTree*)clustFile->Get("o2sim");
+    TTree* clustersRof = (TTree*)clustFile->Get("ITSClustersROF");
+
+    //Read all clusters to a buffer
+    std::vector<itsmft::Cluster>* clusArr = nullptr;
+    clusters->SetBranchAddress("ITSCluster", &clusArr);
+    clusters->GetEntry(0);
+
+    //Read all cluster RO frames to a buffer
+    std::vector<itsmft::ROFRecord>* clusterROFrames = nullptr;
+    clustersRof->SetBranchAddress("ITSClustersROF", &clusterROFrames);
+    clustersRof->GetEntry(0);
+
+    auto currentClusterROF = clusterROFrames->at(event);
+
+    int first, last;
+    first = currentClusterROF.getROFEntry().getIndex();
+    last = first + currentClusterROF.getNROFEntries();
+
+    gsl::span<itsmft::Cluster> mClusters = gsl::make_span(&(*clusArr)[first], last - first);
```

Comment:
Here, an ITS RO frame is treated as an "event" for the visualisation.  This will work for the ITS standalone.  But it is not clear how this interact with the visualisation of other detectors, the TPC in particular...